exchangelib/protocol.py

```diff
@@ -51,7 +52,7 @@ class BaseProtocol(object):
     # The adapter class to use for HTTP requests. Override this if you need e.g. proxy support or specific TLS versions
     HTTP_ADAPTER_CLS = requests.adapters.HTTPAdapter
 
-    def __init__(self, service_endpoint, credentials, auth_type):
+    def __init__(self, service_endpoint, credentials, auth_type, is_service_account=False):
```

Comment:
Can we avoid introducing Nylas-specific concepts into exchangelib? At some point, we will definitely want to switch away from our fork and use the upstream version -- and when that happens, it will be more difficult to make the switch if we have to pull out Nylas-specific concepts.

Maybe we could just set the `SESSION_POOLSIZE` value on the `Protocol` objects for the accounts we want to change?

Reply:
I agree, but I don't see a way to set `SESSION_POOLSIZE` on a specific instance of `Protocol` if it sets up the session and thread pools in the `__init__`. Do you have a suggestion?

Reply:
Do we have a way to access the Protocol object from the Nylas code? If so, we can just assign to `SESSION_POOLSIZE` on the object. There are no private variables in Python, everything is public.

Reply:
Yes. But how do we set a class (not instance) attribute and guarantee it won't affect other instances created?

We could potentially move this session pool setup into a separate method that can take a `pool_size` kwarg that defaults to `SESSION_POOLSIZE` and call it in `__init__`. Then, when the instance is created for service accounts, we can call that method again from our code with the updated pool size to overwrite the existing pools. 

Another option is to switch out `is_service_account` with a `pool_size` kwarg